ingestion/src/metadata/utils/secrets/azure_kv_secrets_manager.py

```diff
@@ -0,0 +1,72 @@
+#  Copyright 2022 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""
+Abstract class for AWS based secrets manager implementations
+"""
+import traceback
+from abc import ABC
+from typing import Optional
+
+from azure.identity import ClientSecretCredential, DefaultAzureCredential
+from azure.keyvault.secrets import KeyVaultSecret, SecretClient
+
+from metadata.generated.schema.security.secrets.secretsManagerProvider import (
+    SecretsManagerProvider,
+)
+from metadata.utils.logger import utils_logger
+from metadata.utils.secrets.external_secrets_manager import ExternalSecretsManager
+
+logger = utils_logger()
+
+NULL_VALUE = "null"
+
+
+class AzureKVSecretsManager(ExternalSecretsManager, ABC):
+    """
+    Azure Key Vault Secrets Manager class
+    """
+
+    def __init__(
+        self,
+        credentials: Optional["AzureCredentials"],
+    ):
+        super().__init__(SecretsManagerProvider.azure_kv)
```

Comment:
I think we can just do 

```
self.provider = SecretsManagerProvider.azure_kv
```

To make it more explicit than call the parent `__init__` since we don't do anything else in the parent `__init__`. WDYT?

Reply:
fair

Reply:
mh actually it might be easier to miss if we don't have it here and just have it as a line we need to remember to add. The IDE / lint would still point us that the class is not calling the super().__init__()